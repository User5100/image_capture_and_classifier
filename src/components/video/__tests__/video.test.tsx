import React from "react";
import { render } from "react-testing-library";
import renderer from "react-test-renderer";
import UI from "../ui";

jest.mock("../helpers", () => {
  return {
    getMediaStream: jest.fn(() => ({})),
  };
});

describe("Video", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<UI />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Displays the correct video title", () => {
    const { getByTestId } = render(<UI />);
    expect(getByTestId("video-title")).toHaveTextContent("Video Title");
  });

  test("Displays the correct video button", () => {
    const { getByTestId } = render(<UI />);
    expect(getByTestId("video-button")).toHaveTextContent("Open Camera");
  });

  test("Displays the 'Take Photo' video button", () => {
    const { getByText } = render(<UI />);
    expect(getByText(/Take Photo/i)).toBeTruthy();
  });

  test("Renders a canvas element", () => {
    const { getByTestId } = render(<UI />);
    expect(getByTestId("canvas")).toBeTruthy();
  });
});
