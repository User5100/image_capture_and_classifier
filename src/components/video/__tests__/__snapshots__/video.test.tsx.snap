// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders correctly 1`] = `
<section>
  <h2
    data-testid="video-title"
  >
    Video Title
  </h2>
  <video
    autoPlay={true}
    className="video"
    data-testid="video"
    playsInline={true}
  />
  <canvas
    className="canvas"
    data-testid="canvas"
  />
  <button
    data-testid="video-button"
    onClick={[Function]}
  >
    Open Camera
  </button>
  <button>
    Take photo
  </button>
</section>
`;
